services/brig/src/Brig/Provider/API.hs

```diff
@@ -509,10 +507,15 @@ getServiceProfile (pid ::: sid) = do
     s <- DB.lookupServiceProfile pid sid >>= maybeServiceNotFound
     return (json s)
 
-listServiceProfilesByTag :: QueryAnyTags 1 3 ::: Maybe Name ::: Range 10 100 Int32 -> Handler Response
-listServiceProfilesByTag (tags ::: start ::: size) = do
+searchServiceProfiles :: Maybe (QueryAnyTags 1 3) ::: Maybe Name ::: Range 10 100 Int32 -> Handler Response
+searchServiceProfiles (Nothing ::: Just start ::: size) = do
+    ss <- DB.paginateServiceNames start (fromRange size) =<< setProviderSearchFilter <$> view settings
```

Comment:
`start` can be `Just ""` in which case the cql lookup won't work (return no results). Or is there code handling the empty string elsewhere?

Reply:
Right... this was overlooked. I thought the `FromByteString` instance checked the size as well but only the `FromJSON` does.
I think the right here is actually to use a `Range 1 128 Text` type, which also solves what you described [earlier on]( 
https://github.com/wireapp/wire-server/pull/197/files/96f4cf6d190f5e43d498b96aeb8be01b7aa3df9a#r167524840)